src/vm/i386/asmhelpers.S

```diff
@@ -1162,20 +1162,30 @@ NESTED_END ResolveWorkerChainLookupAsmStub, _TEXT
 NESTED_ENTRY BackPatchWorkerAsmStub, _TEXT, NoHandler
     PROLOG_BEG
     PROLOG_PUSH eax // it may contain siteAddrForRegisterIndirect
+    PROLOG_PUSH ebx
     PROLOG_PUSH ecx
     PROLOG_PUSH edx
     PROLOG_END
 
-    sub     esp, 4     //  for 16 bytes align
+    // Align Stack Frame
+    mov     ebx, esp
+    and     esp, -16
+
+    // Call BackPatchWorkerStaticStub
+    sub     esp, 8     //  16-bytes align padding
```

Comment:
Sorry for late response. Here is the stack trace when hitting mislaigned stack frame:
```
(gdb) bt
#0  VirtualCallStubManager::BackPatchWorkerStatic (returnAddress=4105930115, siteAddrForRegisterIndirect=4105930208) at /home/parjong/projects/dotnet/coreclr/src/vm/virtualcallstub.cpp:1633
#1  0xf739dabf in BackPatchWorkerStaticStub (returnAddr=4105930115, siteAddrForRegisterIndirect=4105930208) at /home/parjong/projects/dotnet/coreclr/src/vm/virtualcallstub.cpp:4054
#2  0xf75275c3 in BackPatchWorkerAsmStub () from /home/parjong/projects/dotnet/dotnet-overlay/Linux.x86.Checked.Checked/libcoreclr.so
#3  0xf690a05f in ?? ()
#4  0xf4bb8523 in ?? ()
#5  0xf4bb844c in ?? ()
#6  0xf75271f1 in CallDescrWorkerInternal () from /home/parjong/projects/dotnet/dotnet-overlay/Linux.x86.Checked.Checked/libcoreclr.so
#7  0xf73b7d4f in CallDescrWorker (pCallDescrData=<optimized out>) at /home/parjong/projects/dotnet/coreclr/src/vm/callhelpers.cpp:135
#8  0xf73b7be8 in CallDescrWorkerWithHandler (pCallDescrData=<optimized out>, fCriticalCall=<optimized out>) at /home/parjong/projects/dotnet/coreclr/src/vm/callhelpers.cpp:78
#9  0xf73b8ca5 in MethodDescCallSite::CallTargetWorker (this=<optimized out>, pArguments=<optimized out>, pReturnValue=<optimized out>, cbReturnValue=<optimized out>)
    at /home/parjong/projects/dotnet/coreclr/src/vm/callhelpers.cpp:645
#10 0xf75570c6 in Call_RetArgSlot (this=<optimized out>, pArguments=0x3ca8bf) at /home/parjong/projects/dotnet/coreclr/src/vm/callhelpers.h:433
#11 operator() (this=<optimized out>, pParam=<optimized out>) at /home/parjong/projects/dotnet/coreclr/src/vm/assembly.cpp:1851
#12 operator() (this=<optimized out>, __EXparam=<optimized out>) at /home/parjong/projects/dotnet/coreclr/src/vm/assembly.cpp:1865
#13 RunMain (pFD=<optimized out>, numSkipArgs=<optimized out>, piRetVal=<optimized out>, stringArgs=<optimized out>) at /home/parjong/projects/dotnet/coreclr/src/vm/assembly.cpp:1865
#14 0xf75574d9 in Assembly::ExecuteMainMethod (this=<optimized out>, stringArgs=0xf4bb85e0, waitForOtherThreads=-158254204) at /home/parjong/projects/dotnet/coreclr/src/vm/assembly.cpp:1947
#15 0xf7272781 in CorHost2::ExecuteAssembly (this=<optimized out>, dwAppDomainId=1, pwzAssemblyPath=<optimized out>, argc=<optimized out>, argv=<optimized out>, pReturnValue=0xf2a3058c)
    at /home/parjong/projects/dotnet/coreclr/src/vm/corhost.cpp:501
#16 0xf72390d8 in coreclr_execute_assembly (hostHandle=<optimized out>, domainId=<optimized out>, argc=<optimized out>, argv=<optimized out>, managedAssemblyPath=<optimized out>, exitCode=<optimized out>)
    at /home/parjong/projects/dotnet/coreclr/src/dlls/mscoree/unixinterface.cpp:376
#17 0x0804a59e in ExecuteManagedAssembly (currentExeAbsolutePath=<optimized out>, clrFilesAbsolutePath=<optimized out>, managedAssemblyAbsolutePath=<optimized out>, managedAssemblyArgc=-224197236,
    managedAssemblyArgv=0xf4bb85e0) at /home/parjong/projects/dotnet/coreclr/src/coreclr/hosts/unixcoreruncommon/coreruncommon.cpp:405
#18 0x0804981c in corerun (argc=2, argv=0xffffd724) at /home/parjong/projects/dotnet/coreclr/src/coreclr/hosts/unixcorerun/corerun.cpp:149
#19 0x08049acf in main (argc=2, argv=0xffffd724) at /home/parjong/projects/dotnet/coreclr/src/coreclr/hosts/unixcorerun/corerun.cpp:161
(gdb) info frame 2
Stack frame at 0xffffca4c:
 eip = 0xf75275c3 in BackPatchWorkerAsmStub; saved eip = 0xf690a05f
 called by frame at 0xffffca70, caller of frame at 0xffffca2c
 Arglist at 0xffffca44, args:
 Locals at 0xffffca44, Previous frame's sp is 0xffffca4c
 Saved registers:
  eax at 0xffffca40, ecx at 0xffffca3c, edx at 0xffffca38, ebp at 0xffffca44, eip at 0xffffca48
(gdb) info frame 3
Stack frame at 0xffffca70:
 eip = 0xf690a05f; saved eip = 0xf4bb8523
 called by frame at 0xffffca90, caller of frame at 0xffffca4c
 Arglist at 0xffffca68, args:
 Locals at 0xffffca68, Previous frame's sp is 0xffffca70
 Saved registers:
  ebp at 0xffffca68, eip at 0xffffca6c
```